### PR TITLE
1065 add offline mode

### PR DESCRIPTION
**Pull request by Karina5005**
https://github.com/objectionary/eo/issues/1065

### CODE REVIEW

**Comment by Karina5005** [comment]
@mximp I have a problem. 
As discussed, I skipped PullMojo in case of no internet connection. But because of this, there is a problem with the fibonacci test and eo-runtime tests. The eo base classes are not loaded (which is expected if there is no internet connection), so a compilation error occurs. 
In the case of using the skip option (`mvn -Deo.skip=true clean install`), tests in eo-runtime also fall, so maybe this is normal behavior?

**Comment by yegor256** [comment]
@MikhailLipanin please, review this one

**Comment by mximp** [comment]
@Karina5005 for fibonacci test it's probably fine - since it's part of `eo-maven-plugin` module and eo runtime is external to it. 
But why it fails for `eo-runtime` all sources are available right in the module? Could you please advise what files are missing?

**Review by mximp — CHANGES_REQUESTED** [review]

**Comment by Karina5005** [comment]
@mximp Files not found when compiling. For example, here are a few:
```
[ERROR] /home/karina/eo/eo-runtime/src/main/java/org/eolang/Data.java:[30,22] cannot find symbol
  symbol:   class EOfloat
  location: package EOorg.EOeolang
[ERROR] /home/karina/eo/eo-runtime/src/main/java/org/eolang/Data.java:[31,22] cannot find symbol
  symbol:   class EOint
  location: package EOorg.EOeolang
[ERROR] /home/karina/eo/eo-runtime/src/main/java/org/eolang/Data.java:[32,22] cannot find symbol
  symbol:   class EOstring
  location: package EOorg.EOeolang
```

**Comment by mximp** [comment]
@Karina5005 would the tests fail even objects are located in `~\.eo` folder? EO compilation and tests require external objects. If there is no access to web and required objects were not previously downloaded it's logical that the build is failing.

**Review by mximp — APPROVED** [review]

**Review by mximp — APPROVED** [review]

File: eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java [hunk 1]
```diff
@@ -154,4 +157,22 @@ void assemblesNotFailWithFailOnErrorFlag(@TempDir final Path temp) throws Except
         );
     }
 
+    /**
+     * Check the internet connection.
+     *
+     * @return Is the internet connection available
+     */
+    private static boolean netIsAvailable() {
```
[the remark this answers is not in the document]
Comment (Karina5005) [reply eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java]: @mximp I made it a static method of the SafeMojo class, it seems logical to me, perhaps in the future this function will be needed in other Mojos. I'm right?

File: eo-maven-plugin/src/it/fibonacci/verify.groovy [hunk 2]
```diff
@@ -22,18 +22,39 @@
  * SOFTWARE.
  */
 
+/**
+ * Check the internet connection.
+ *
+ * @return Is the internet connection available
+ */
+private static boolean netIsAvailable() {
```
[the remark this answers is not in the document]
Comment (Karina5005) [reply eo-maven-plugin/src/it/fibonacci/verify.groovy]: fixed

File: eo-maven-plugin/src/main/java/org/eolang/maven/PullMojo.java [hunk 3]
```diff
@@ -171,4 +179,22 @@ private Path pull(final String name) throws IOException {
         return src;
     }
 
+    /**
+     * Check the internet connection.
+     *
+     * @return Is the internet connection available
+     */
+    private static boolean netIsAvailable() {
```
[the remark this answers is not in the document]
Comment (Karina5005) [reply eo-maven-plugin/src/main/java/org/eolang/maven/PullMojo.java]: fixed

File: eo-maven-plugin/src/main/java/org/eolang/maven/PullMojo.java [hunk 4]
```diff
@@ -171,4 +179,22 @@ private Path pull(final String name) throws IOException {
...
+        boolean available = true;
+        try {
+            final URL url = new URL("http://www.google.com");
```
[the remark this answers is not in the document]
Comment (Karina5005) [reply eo-maven-plugin/src/main/java/org/eolang/maven/PullMojo.java]: fixed

**Comment by mximp** [comment]
@yegor256 can we please merge this

**Comment by Karina5005** [comment]
@mximp I'm not sure I understand how to run it so that the downloaded files are not deleted.
I tried like this:
```
mvn clean install
mvn test
```
 But the files are not found.
```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/karina/eo/eo-runtime/target/generated-sources/EOorg/EOeolang/EOmath/EOnumber.java:[366,21] cannot find symbol
  symbol:   class EOnumber$EOpow
  location: class EOorg.EOeolang.EOmath.EOnumber
[ERROR] /home/karina/eo/eo-runtime/target/generated-sources/EOorg/EOeolang/EOmath/EOnumber.java:[371,21] cannot find symbol
  symbol:   class EOnumber$EOas_float
  location: class EOorg.EOeolang.EOmath.EOnumber
[ERROR] /home/karina/eo/eo-runtime/target/generated-sources/EOorg/EOeolang/EOmath/EOnumber.java:[376,21] cannot find symbol
  symbol:   class EOnumber$EOas_int
  location: class EOorg.EOeolang.EOmath.EOnumber
[ERROR] /home/karina/eo/eo-runtime/target/generated-sources/EOorg/EOeolang/EOmath/EOnumber.java:[381,21] cannot find symbol
  symbol:   class EOnumber$EOas_string
  location: class EOorg.EOeolang.EOmath.EOnumber
[ERROR] /home/karina/eo/eo-runtime/target/generated-sources/EOorg/EOeolang/EOmath/EOangle.java:[83,21] cannot find symbol
  symbol:   class EOangle$EOsin
  location: class EOorg.EOeolang.EOmath.EOangle
[ERROR] /home/karina/eo/eo-runtime/target/generated-sources/EOorg/EOeolang/EOmath/EOangle.java:[88,21] cannot find symbol
  symbol:   class EOangle$EOcos
  location: class EOorg.EOeolang.EOmath.EOangle
[ERROR] /home/karina/eo/eo-runtime/target/generated-sources/EOorg/EOeolang/EOmath/EOrandom.java:[48,21] cannot find symbol
  symbol:   class EOrandom$EOpseudo
  location: class EOorg.EOeolang.EOmath.EOrandom
[ERROR] /home/karina/eo/eo-runtime/target/generated-sources/EOorg/EOeolang/EOcollections/EOmultimap.java:[206,21] cannot find symbol
  symbol:   class EOmultimap$EOrebuild
  location: class EOorg.EOeolang.EOcollections.EOmultimap
```

**Comment by mximp** [comment]
@Karina5005 are you trying to build it in parent eo module?
Can you please check your `~/.eo/` folder for those EO files?

**Comment by yegor256** [comment]
@Karina5005 a general rules is this: if something doesn't work as expected in the `master` branch, make a new ticket about it and wait. When the issue is resolved in the `master` branch, continue the development in your own branch.

**Comment by yegor256** [comment]
@Karina5005 indeed, this scenario fails in `master` (I just checked):

```
$ mvn clean install
$ mvn test
```

I suggest you to make a new ticket about it.

**Comment by yegor256** [comment]
@Karina5005 what's up with this one?

**Comment by Karina5005** [comment]
@yegor256 The ticket https://github.com/objectionary/eo/issues/1130 is still open, so I can't test properly.

**Comment by yegor256** [comment]
@mximp can you please pick this up and finish it somehow please?

**Comment by mximp** [comment]
@yegor256 I will recreate PR

**Comment by mximp** [comment]
recreated in https://github.com/objectionary/eo/pull/1550

**Pull request closed without merging**